includes/acl/math/scalar_packing.h

```diff
@@ -36,6 +36,37 @@ ACL_IMPL_FILE_PRAGMA_PUSH
 
 namespace acl
 {
+
+#ifdef ACL_PRECISION_BOOST
+
+	// We don't support [0.0F, 1.0F].
+	// It's imprecise, and (0.5F, 1.0F) is incapable of representing the error term of 24 bit quantization.
+	// We support [-0.5F, 0.5F] instead.
+	inline uint32_t pack_scalar_signed_normalized(float input, uint32_t num_bits)
+	{
+		ACL_ASSERT(num_bits > 0, "Attempting to pack on too few bits");
+		ACL_ASSERT(num_bits < 25, "Attempting to pack on too many bits");
+		ACL_ASSERT(input >= -0.5F && input <= 0.5F, "Expected signed normalized input value: %f", input);
+		const uint32_t num_values = 1 << num_bits;
+		const float max_value = rtm::scalar_safe_to_float(num_values);
+		const float mid_value = 0.5F * max_value;
+		return std::min(static_cast<uint32_t>(rtm::scalar_floor(input * max_value) + mid_value), num_values - 1);
+	}
+	
+	inline float unpack_scalar_signed_normalized(uint32_t input, uint32_t num_bits)
+	{
+		ACL_ASSERT(num_bits > 0, "Attempting to unpack from too few bits");
+		ACL_ASSERT(num_bits < 25, "Attempting to unpack from too many bits");
+		const uint32_t num_values = 1 << num_bits;
+		ACL_ASSERT(input < num_values, "Input value too large: %ull", input);
+		const float max_value = rtm::scalar_safe_to_float(num_values);
+		const float inv_max_value = 1.0F / max_value;
+		const float mid_value = (0.5F * max_value) - 0.5F;
+		return (input - mid_value) * inv_max_value;
```

Comment:
Here for 8 bits we'd use 256 and for 16 bits 65536 which differs from the code earlier. Why?